modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/CreateSystemViewCommand.java

```diff
@@ -0,0 +1,207 @@
+
+import static java.util.stream.Collectors.toList;
+import static org.apache.ignite.internal.catalog.CatalogParamsValidationUtils.validateColumnParams;
+import static org.apache.ignite.internal.catalog.CatalogParamsValidationUtils.validateIdentifier;
+import static org.apache.ignite.internal.util.CollectionUtils.nullOrEmpty;
+import static org.apache.ignite.lang.IgniteStringFormatter.format;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.apache.ignite.internal.catalog.Catalog;
+import org.apache.ignite.internal.catalog.CatalogManager;
+import org.apache.ignite.internal.catalog.CatalogParamsValidationUtils;
+import org.apache.ignite.internal.catalog.CatalogValidationException;
+import org.apache.ignite.internal.catalog.descriptors.CatalogSchemaDescriptor;
+import org.apache.ignite.internal.catalog.descriptors.CatalogSystemViewDescriptor;
+import org.apache.ignite.internal.catalog.descriptors.CatalogSystemViewDescriptor.SystemViewType;
+import org.apache.ignite.internal.catalog.descriptors.CatalogTableColumnDescriptor;
+import org.apache.ignite.internal.catalog.storage.NewSystemViewEntry;
+import org.apache.ignite.internal.catalog.storage.ObjectIdGenUpdateEntry;
+import org.apache.ignite.internal.catalog.storage.UpdateEntry;
+
+/**
+ * Create system view command - creates or replaces a system view.
+ *
+ * <p>If a system with with the given name does not exists, this command adds this system view to {@code SYSTEM} schema.
+ * If a system with with the given name exists, this command replaces the existing system view.
+ */
+public class CreateSystemViewCommand extends AbstractCatalogCommand {
+
+    private final String name;
+
+    private final List<ColumnParams> columns;
+
+    private final SystemViewType systemViewType;
+
+    /**
+     * Constructor.
+     *
+     * @param name View name.
+     * @param columns List of view columns.
+     * @param systemViewType View type.
+     */
+    CreateSystemViewCommand(String name, List<ColumnParams> columns, SystemViewType systemViewType) {
+        this.name = name;
+        this.columns = columns;
+        this.systemViewType = systemViewType;
+
+        validate();
+    }
+
+    /**
+     * Returns a view name.
+     *
+     * @return View name.
+     */
+    public String name() {
+        return name;
+    }
+
+    /**
+     * Returns a list of view columns.
+     *
+     * @return List of view columns.
+     */
+    public List<ColumnParams> columns() {
+        return columns;
+    }
+
+
+    /**
+     * Returns a view type.
+     *
+     * @return View type.
+     */
+    public SystemViewType systemViewType() {
+        return systemViewType;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<UpdateEntry> get(Catalog catalog) {
+        int id = catalog.objectIdGenState();
+
+        CatalogSchemaDescriptor systemSchema = catalog.schema(CatalogManager.SYSTEM_SCHEMA_NAME);
```

Comment:
I believe, it's better to throw validation exception here if schema is not found rather then NPE a few lines later